src/app/_components/Event/EventFilter.js

```diff
@@ -0,0 +1,179 @@
+'use client';
+
+import { useState, useEffect } from 'react';
+import { useInfiniteQuery } from '@tanstack/react-query';
+import { useInView } from 'react-intersection-observer';
+import { useSearchParams } from 'next/navigation';
+import { CircularProgress } from '@mui/material';
+import CheckMark from '@icons/check-mark.svg';
+import Search from '@icons/search.svg';
+import { allEvents, useEventSetParam } from '@hooks';
+import { cn } from '@/utils/cn';
+import { PillButton } from '../PillButton';
+import { buttonColorVariant } from '../PillButton/style';
+import { NoInfoToShow } from '../NoInfoToShow';
+import { EventCard } from './Card';
+import { monthsAhead, monthNames } from './config';
+
+export function EventFilter() {
+  const [months, setMonths] = useState([]);
+  const [activeIndex, setActiveIndex] = useState(0);
+  const [activeMonth, setActiveMonth] = useState(months[0]);
+  const [activeMonthNumber, setActiveMonthNumber] = useState('');
+  const [hoveredIndexes, setHoveredIndexes] = useState(new Array(months.length));
+
+  const currentMonth = new Date().getMonth() + 1;
+
+  const searchParams = useSearchParams();
+  const { addParam, deleteParam } = useEventSetParam('month');
+
+  const { ref, inView } = useInView();
+
+  const { semiorange } = buttonColorVariant.eventFilter;
+
+  const activeButtonStyles = cn({
+    'pointer-events-none border-secondary-300 bg-secondary-300 font-semibold text-gray-900': semiorange.active,
+  });
+
+  function selectButtonIcon(index) {
+    if (activeIndex === index && semiorange.active) {
+      return <CheckMark key={`checkmark+${index}`} className="h-4 w-4 transition-all" />;
+    }
+    if (hoveredIndexes[index] && activeIndex !== index && (semiorange.hover || semiorange.focused)) {
+      return <Search key={`searchicon+${index}`} className="h-4 w-4 transition-all " />;
+    }
+    return null;
+  }
+
+  // Get 6 basic month names for filters, starting from current month
+  useEffect(() => {
+    const getNextMonths = () => {
+      const nextMonths = [];
+      for (let i = 0; i <= monthsAhead; i += 1) {
+        const nextMonthIndex = (currentMonth + i - 1) % 12;
+        const nextMonthName = monthNames[nextMonthIndex];
+        setActiveMonthNumber(monthNames[i] === currentMonth ? monthNames[i] : '');
+        nextMonths.push(nextMonthName);
+      }
+      return nextMonths;
+    };
+    const nextMonths = getNextMonths();
+    setMonths(nextMonths);
+    // eslint-disable-next-line
+  }, [currentMonth]);
```

Comment:
@Yivgha there is multiple things 
1. I don't see a reason those logic to be an effect. It can be calculated statically outside of the component
2. logic is too complicated for such case and it leads for mistakes. monthNames[i] is object, when currentMonth is a number, so comparing them with === is unusfull. Let's rewrite this function with .slice method of arrays. (you can check that monthsAhead is out of the length of the array and make concatination of 2 slices).
3. We should rid all of the states and make them static outside of the component, they are not depend on any reactive things

Reply:
changed